### PR TITLE
Ruinian/nlu eval script

### PR DESCRIPTION
**Pull request by ruinianxu**
# Description

This PR is for rewriting NLU evaluating script. Currently it supports two models: query and eval model. The query model takes a sentence as input while eval model measures the accuracy of the given dataset. Additionally, the unit test ```test_y_print_parsing_report.py``` includes flag to enable whether loading real nsp model.

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [x] Refactor
- [x] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [x] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

## Before and After

Before: 
- ```test_model_script.py``` only supports querying the input sentence. 
- ```test_y_print_parsing_report.py``` loads real nsp model to parse some commonly used queries
After: 
- ```test_model_script.py``` supports both query and eval models.
- ```test_y_print_parsing_report.py``` supports a flag for whether loading real nsp model for the purpose of resolving the issue of taking too long to run in CircleCI.

# Testing

The NLU model downloaded via main branch doesn't work since grammar increases as dataset grows. To run the evaluation script, you need to fetch new artifacts/models.

```
python droidlet/tools/artifact_scripts/fetch_artifacts_from_aws.py --agent_name craftassist --artifact_name models --model_name nlu --checksum_file nlu.txt
python droidlet/tools/artifact_scripts/fetch_artifacts_from_aws.py --agent_name craftassist --artifact_name datasets --checksum_file datasets.txt
```

Run it in ipython
```
ipython
from tools.nsp_scripts.test_model_script import *
args = argument_parse("")
model, tokenizer = model_configure(args)
dataset = dataset_configure(args, tokenizer)
query_model("hello", args, model, tokenizer, dataset)
eval_model(args, model, tokenizer, dataset)
```

# Checklist:

- [x] I have performed manual end-to-end testing of the feature in my environment.
- [ ] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added relevant collaborators to review the PR before merge.
- [ ] [Polymetis only] I ran on hardware (1) all scripts in `tests/scripts`, (2) asv benchmarks.

### CODE REVIEW

**Review by kavyasrinet — COMMENTED** [review]
Thank you for adding detailed description in the PR, that's really useful.
The changes look good to me, but let's have @aszlam stamp it before we merge.

File: droidlet/perception/semantic_parsing/tests/test_y_print_parsing_report.py [hunk 1]
```diff
@@ -1185,14 +1186,23 @@ def compare_full_dictionaries(d1, d2):
...
+@pytest.mark.usefixtures("flag_m")
 class TestDialogueManager(unittest.TestCase):
     def setUp(self):
         opts = MockOpt()
         opts.nsp_data_dir = TTAD_BERT_DATA_DIR
         opts.ground_truth_data_dir = GROUND_TRUTH_DATA_DIR
         opts.nsp_models_dir = TTAD_MODEL_DIR
         opts.no_ground_truth = False
-        self.chat_parser = NSPQuerier(opts=opts)
+        if self.flag_m == "True":
```
Comment (snyxan) [inline droidlet/perception/semantic_parsing/tests/test_y_print_parsing_report.py:1197]: sorry I don't know the `pytest.mark.usefixtures` very well, can we use the following here:
`opts.flat_m = self.flag_m`?

Comment (kavyasrinet) [reply droidlet/perception/semantic_parsing/tests/test_y_print_parsing_report.py:1197]: Thanks @snyxan , I agree with this comment as well

File: droidlet/perception/semantic_parsing/tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import pytest
+
+
+def pytest_addoption(parser):
+    parser.addoption("--flag_m", action="store", default="Whether to load real NSP model")
```
[the remark this answers is not in the document]
Comment (ruinianxu) [reply droidlet/perception/semantic_parsing/tests/conftest.py]: update to ```flag_load_nsp_model```

File: droidlet/perception/semantic_parsing/tests/test_y_print_parsing_report.py (hunk 1 above)
Comment (ruinianxu) [reply droidlet/perception/semantic_parsing/tests/test_y_print_parsing_report.py]: I update the type of flag to store_true. Thanks for comment.

**Comment by ruinianxu** [comment]
@aszlam 
Could you please stamp the PR if it looks fine?

**Review by aszlam — APPROVED** [review]

**Pull request merged**